test/tangle-prune.test.js

```diff
@@ -0,0 +1,87 @@
+// SPDX-FileCopyrightText: 2022 Mix Irving
+//
+// SPDX-License-Identifier: LGPL-3.0-only
+
+const test = require('tape')
+//const bipf = require('bipf')
+const Testbot = require('./helpers/testbot')
+const prunePublish = require('../lib/prune-publish')
+
+//const chars = 'abcABC123=+? '.split('')
+//const encodedLength = (obj) => bipf.encodingLength(obj)
+//const randomChar = () => chars.sort(() => (Math.random() < 0.5 ? -1 : +1))[0]
+//const randomText = (length) => {
+//  let output = ''
+//  while (output.length < length) output += randomChar()
+//  return output
+//}
+
+test('prune a message with way too big `previous`', (t) => {
+  const ssb = Testbot()
+  const ssbId = ssb.id
+
+  ssb.tribes2.create(null, (err, group) => {
+    t.error(err, 'no err on group create')
+
+    const msgId = '%RDORgMCjmL6vs51nR4bn0LWNe6wkBfbRJulSdOJsmwg=.sha256'
+    const content = (prevCount, numRecps) => ({
+      type: 'post',
+      text: 'hello!',
+      recps: [group.id, new Array(numRecps - 1).fill(ssbId)],
+      tangles: {
+        group: {
+          root: msgId,
+          previous: new Array(prevCount).fill(msgId),
+        },
+      },
+    })
+
+    //console.time('prune')
+    prunePublish(ssb, content(4000, 16), (err, encMsg16) => {
+      //console.timeEnd('prune')
```

Comment:
approx 3-6 ms. pruned from 4000 to 63 entries

doing this 5k times would take approx 25s

Reply:
nice